video/image_writer.c

```diff
@@ -262,11 +309,188 @@ static bool write_jpeg(struct image_writer_ctx *ctx, mp_image_t *image, FILE *fp
 
     jpeg_destroy_compress(&cinfo);
 
-    return true;
+    return !fclose(fp);
 }
 
 #endif
 
+#if HAVE_AVIF_MUXER
+
+#define HANDLE_ERROR(expr,...) if (expr) do { \
+    MP_ERR(ctx, __VA_ARGS__); \
+    goto free_data; \
+} while(0)
```

Comment:
Not a fan of this macro personally. Maybe if you were doing something really long winded, but `if (!something)` + `MP_ERR("message")` + `goto err` is perfectly fine and that pattern is already everywhere in mpv code.

Reply:
I just like the look of
```
tmp = function();
HANDLE(cond, message);
```
but if you say so, I can inline that macro.

Reply:
Well no I don't want this to be a function either. It should just be written out explicitly. I know it's repetitive but that's the current convention.

Reply:
kk, done.